pytorch_lightning/trainer/trainer.py

```diff
@@ -1072,13 +1093,16 @@ def _run_sanity_check(self, ref_model):
             # restore the previous stage when the sanity check if finished
             self.state.stage = stage
 
-    def __load_ckpt_weights(self, ckpt_path: Optional[str]) -> Optional[str]:
-        if ckpt_path is None:
-            return
+    def __set_ckpt_path(self, ckpt_path: Optional[str], model_provided: bool, model_connected: bool) -> Optional[str]:
+        if model_provided and (ckpt_path in ('best', None)):
+            return  # use passed model to function without loading weights
```

Comment:
I think passing `best` should still load the best model. Otherwise it should print a warning so the users get an indication about passing `trainer.model_checkpoint.best_checkpoint_path` themselves.

Suggestion for unnecessary parentheses:

```suggestion
        if model_provided and ckpt_path in ('best', None):
            return  # use passed model to function without loading weights
```

Reply:
I agree here however this adds complication as to what should be the default, and adds confusion:

```
ckpt_path by default is None

if model is provided and ckpt_path == None then use model
if model is provided and ckpt_path == 'best' then load best model
if model is connected to trainer and ckpt_path == None then load best model
if model is  connected to trainer  and ckpt_path == 'best' then load best model
```